micro-rdk/src/common/conn/viam.rs

```diff
@@ -1355,6 +1363,18 @@ mod tests {
             let _fake_server_task =
                 cloned_exec.spawn(async move { run_fake_app_server(other_clone, app).await });
             let _task = cloned_exec.spawn(async move {
+                let (lock, cvar) = &*pair;
+                for _ in 0..3 {
+                    // lock within loop to drop mutex guard between (a)waits
```

Comment:
refer to [`On using std::sync::Mutex and tokio::sync::Mutex`](https://tokio.rs/tokio/tutorial/shared-state).